applications/mne_scan/mne_scan/main.cpp

```diff
@@ -122,18 +122,23 @@ int main(int argc, char *argv[])
 
     QPixmap pixmap(":/images/splashscreen.png");
     MainSplashScreen::SPtr splashscreen(new MainSplashScreen(pixmap));
+
+    MainWindow mainWin;
+    mainWin.setSplashScreen(*splashscreen);
```

Comment:
Why did you choose to share the splash screen with the main window itself? Why not use a signal in MainWindow which forwards the plugin message to the splash screen living outside of MainWindow? This way the two do not need to know about each other.

Reply:
Because I don't want the pluginmanager to know anything about splashing windows. Because they live in different parts of my head? I believe that mainwindow is a bad name for a configuration interface class. There, it is ok to have knowledge of the pluginmanager (which is how it is done right now) and it seems ok to me, to have some reference to hold where the splash window is... should there be one.

Reply:
I aggree. However, with my proposed idea the PluginManager would not have to know about the SplashScreen if you would forward the signal from the pluginmanager through the mainwindow to the main method. Only main() would know about the splash screen.

I do not like the fact that the splash screen is now shared between main() and the MainWindow. Why does the MainWindow need to know about the splash screen at all? And if even if this is absolutley necessary why not putting the splash screen directly into the MainWindow then?

Reply:
I like your idea of keeping splash screen in the MainWindow class. I actually didn't do that just because I didn't want to change the code that much. I think that would actually make a lot of sense.

>However, with my proposed idea the PluginManager would not have to know about the SplashScreen ... 

I honestly forgot about what changes we did online. I tried to find them... but I couldn't, so I started again. But if you want to do it, it will be a nice exercise for me. 
The configuring class (i.e. MainWindow), configures and uses the plugin manager, and if there is a splash screen, connects it to that thing.
PluginManager just broadcasts. That's the idea behind signals, right? PluginManager doesn't know about who's listening. Where do you see PluginManager acknowledging the splash screen?

>Only main() would know about the splash screen.

But that way, why would main need to know if there is such a thing called plugin? It really doesn't belong there.

> I do not like the fact that the splash screen is now shared between main() and the MainWindow. 

First of all, it is only a reference, which can be not configured. Secondly, I agree 100%. 😄 

>if you would forward the signal from the pluginmanager through the mainwindow to the main method

Trying to understand what you mean. Do you mean to have MainWindow act as a relay of signals? But why would you want to do that? Signals are a broadcasting mechanism. Very cool. You want to listen somewhere?... you just tune in. What if you want to add some log class with some slot to which you also want to connect the plugin manager? You can do that now, independently of everything else.

So would you fork and maybe show me? Or through a skype (very trendy these days)?
We could focus on this and the static compilation.

Reply:
yes,  let's discuss this in a Skype.